mlir-tensorrt/tensorrt/lib/TensorRT/IR/TensorRT.cpp

```diff
@@ -2448,6 +2448,45 @@ LogicalResult OpaquePluginOp::reifyResultShapes(
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// ResizeOp
+//===----------------------------------------------------------------------===//
+
+/// Fold `tensorrt.op(..., tensor.cast(x)... )` to `tensorrt.op(..., x, ...)`
+/// if the cast is a generalizing cast (it is removing some static dims of the
+/// type of  `x` and replacing them with dynamic dimensions).
+template <typename OpType>
+struct AbsorbTensorCastOp : public OpRewritePattern<OpType> {
+  using OpRewritePattern<OpType>::OpRewritePattern;
+  LogicalResult matchAndRewrite(OpType op,
+                                PatternRewriter &rewriter) const override {
+    if (auto castOp = op.getInput().template getDefiningOp<tensor::CastOp>()) {
+      if (!isTargetRefinementOfSource(castOp.getType().getShape(),
+                                      castOp.getSource().getType().getShape()))
```

Comment:
Please add condition `castOp.getType().getEncoding() == castOp.getSource().getType().getEncoding()`.

Cast op may change shape dimensions or encoding. If the encoding also changes (in addition to generalizing the shape), we should not perform this change since we lose some information. The likelihood of encountering the situation is low; we  just have to be conservative.

Reply:
Added, thanks.